fms/utils/generation.py

```diff
@@ -55,6 +55,8 @@ def generate(
 
     for _ in range(max_new_tokens):
         input_ids = next_input[:, -max_seq_len:]
+        # Update cacheable data in model
+        model.update_cacheable_data(input_ids, **kwargs)
```

Comment:
hmm... this imposes a fixed API on every model that would use `generate`. we were trying to avoid that. There's also no enforcement, e.g., code in the GPTBigCode PR would now fail here if using this function.

Reply:
I wanted to discuss this actually. I might make it so that if a model has it it runs and if it's not defined it's just ignored, but I can't think of a way to avoid the graph break other than separating this part of the function outside the main forward pass